### ISSUE TITLE
"Generic spaces mismatch" false positive for files using Windows line endings

### ISSUE THREAD

**Issue by sebthom** [antfu/eslint-plugin-antfu#3]
### Describe the bug

Using **@antfu/eslint-config-ts@0.43.1** a definition like
```typescript
type Foo<
  T = true,
  K = false,
> = T
```
in a typescript file with Windows line endings results in the error **error  Generic spaces mismatch  antfu/generic-spacing**.

### Reproduction

see above

### System Info

```Shell
System:
    OS: Windows 10 10.0.19045
    CPU: does-not-matter
    Memory: does-not-matter
  Binaries:
    Node: 18.18.0 - C:\nodejs-lts\current\node.EXE
    npm: 9.8.1 - C:\nodejs-lts\current\npm.CMD
  Browsers:
    Edge: does-not-matter
    Internet Explorer: does-not-matter
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.
